MsCorePkg/Include/Library/MacAddressEmulationPlatformLib.h

```diff
@@ -0,0 +1,53 @@
+/** @file
+
+  Defines the APIs for the MAC Address Emulation Platform library.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+#ifndef _MAC_ADDRESS_EMULATION_PLATFORM_LIB_H_
+#define _MAC_ADDRESS_EMULATION_PLATFORM_LIB_H_
+
+/**
+  Called by the MacAddressEmulationDxe driver's entry function to determine if the platform's MAC emulation is enabled.
```

Comment:
a library api should not speculate or define who will call it when.  This api can specify the requirements on the environment when it is called.